### ISSUE TITLE
img2img+img2noise: artifacts at moderate blend_strength (signal-noise correlation)

### ISSUE THREAD

**Issue by djdarcy** [djdarcy/ComfyUI-Smart-Resolution-Calc#47]
## Summary

`img2img + img2noise` mode produces artifacts (stippling, geometric shapes, posterization) at moderate-to-high `blend_strength` values. The standalone `img2noise` mode works well after the VAE-encode pattern fix (v0.10.1), but the layered mode where the same image serves as both VAE-encoded samples AND noise pattern source has quality issues.

## Problem

When `image_purpose = "img2img + img2noise"`:
- `blend_strength = 0.0` — clean output, identical to standard img2img (confirmed working)
- `blend_strength = 0.3, denoise = 0.3` — subtle stippling artifacts in high-detail areas (grass)
- `blend_strength = 0.9, denoise = 0.3` — washed out, ghostly rendering
- `blend_strength = 0.9, denoise = 0.66` — severe artifacts (geometric shapes, horizontal lines, posterization)

Standard `img2img` at the same denoise/seed produces clean results. Standalone `img2noise` with VAE-encoded pattern works well at blend up to ~0.4.

## Root Cause Analysis

The VAE-encoded image is used for **both** the latent samples (x_0) and the noise pattern source. When the sampler computes:

```
x_t = sqrt(alpha_t) * x_0 + sqrt(1-alpha_t) * shaped_noise
```

Both `x_0` and `shaped_noise` carry the **same** spatial structure (derived from the same VAE encoding). This creates:

1. **Correlation between signal and noise** — the noise reinforces the image's own frequency profile instead of providing independent perturbation. At high blend, the noise is essentially a copy of the image structure.

2. **Double amplification of spatial features** — dark areas (silhouette) get both preserved signal AND correlated noise, pushing them to extremes. High-frequency areas (grass detail) get coherent structure from both sources, which the model renders as texture artifacts.

3. **Non-Gaussian effective noise** — even though `spectral_noise_blend()` normalizes to std=1, the blend of VAE-encoded pattern + Gaussian noise produces a distribution with different higher-order statistics (skewness, kurtosis) than pure Gaussian. The sampler's noise schedule assumes Gaussian — deviations cause over/under-denoising.

## Empirical Results

| blend_strength | denoise | fill_type | Result |
|---------------|---------|-----------|--------|
| 0.0 | 0.3 | DazNoise: Pink | Clean (matches standard img2img) |
| 0.3 | 0.3 | DazNoise: Pink | Mild stippling in high-detail areas |
| 0.9 | 0.3 | DazNoise: Pink | Washed out, ghostly |
| 0.9 | 0.66 | DazNoise: Pink | Severe geometric artifacts |
| 0.25 | 0.66 | DazNoise: Plasma | Decent output (composition influence visible) |

Model: Qwen Image, 19 steps, dpmpp_3m sampler

### 1. Phase Randomization
Keep the amplitude spectrum of the VAE-encoded pattern (preserves spatial structure) but randomize the phases. This breaks pixel-level correlation with samples while preserving "where the blobs are."

```python
F_pattern = torch.fft.rfft2(pattern)
random_phase = torch.angle(torch.randn_like(F_pattern) + 1j * torch.randn_like(F_pattern))
F_decorrelated = torch.abs(F_pattern) * torch.exp(1j * random_phase)
pattern_decorrelated = torch.fft.irfft2(F_decorrelated)
```

### 2. Spatial Offset Decorrelation
VAE-encode a slightly shifted version of the image for the noise pattern (e.g., 2-4 pixel shift, or a small random crop). This preserves large-scale composition but breaks the per-pixel correlation between samples and noise.

### 3. Lower Effective Blend for Layered Mode
The reinforcement effect means blend=0.1 in layered mode has similar composition influence to blend=0.3 in standalone mode. Cap the effective blend or scale it differently:
```python
if use_image_for_latent_encode and use_image_for_noise_shape:
    effective_blend = blend_strength * 0.3  # Scale down for layered reinforcement
```

### 4. Independent Noise Source
Instead of using the same VAE-encoded image for both samples and noise pattern, use a **different** representation:
- VAE-encode the image for samples (standard)
- Use the fill_type noise pattern (not the image) for the noise shaping
- The image still influences via x_0, the fill pattern provides independent spectral structure

### 5. Noise Distribution Analysis
Log the actual distribution statistics (mean, std, skewness, kurtosis) of the shaped noise vs pure Gaussian and compare. This would quantify how far from Gaussian the spectral blend output is.

## Workaround (Current)

Users should keep `blend_strength` very low (0.05-0.15) when using `img2img + img2noise`, or use standalone `img2noise` mode which works well at higher blend values.

## Related

- Issue #42: Epic for spectral fusion
- Issue #34: DazzleKSampler (needed for proper noise integration)
- Issue #45: 2D XY pad widget (would help visualize safe zones)
- `2026-03-20__04-46-47__dev-workflow-img2noise-diagonal-artifacts.md`
- `notes/ideas/2026-03-20__07-00-15__both_img2img-img2noise-quality-issue.md`
